### ISSUE TITLE
Feature Request: Update signup_flow_name of account creation step in use cases to reflect the flow

### ISSUE THREAD

**Issue by ebinnion** [Automattic/wp-calypso#67173]
### What

From the account creation step of the tailored use cases flows, the `signup_flow_name` value is always `account`. It seems more correct if we updated this to reflect the flow that we're in. From testing, `flow` seems to be set as a query arg, so it should be relatively straight forward to pull this out and send it in the user account creation call.

### Why

This would be helpful for our email communications as part of the use cases flows where we need to know the flow that the user came in _before_ the site is created.

**Comment by ebinnion** [comment]
@alshakero  - I went ahead and assigned you as a proxy. I might be able to get to this myself today or tomorrow though.

**Comment by ebinnion** [comment]
After looking into this a bit, I think that it's a side effect of us jumping between stepper and start frameworks. When I add the `user` step to both the `newsletter` and `link-in-bio` flows in start, I get the `signup_flow_name` as expected... but, then I jump back out to the stepper flow when expected. This was the diff that I had locally to get `signup_flow_name` to get sent as expected.

```
❯ git diff
diff --git a/client/landing/stepper/declarative-flow/newsletter.ts b/client/landing/stepper/declarative-flow/newsletter.ts
index f2fc187fc8..fff4962afa 100644
--- a/client/landing/stepper/declarative-flow/newsletter.ts
+++ b/client/landing/stepper/declarative-flow/newsletter.ts
@@ -45,7 +45,7 @@ export const newsletter: Flow = {
                                                return navigate( 'newsletterSetup' );
                                        }
                                        return window.location.replace(
-                                               '/start/account?flowName=newsletter&redirect_to=/setup/newsletterSetup?flow=newsletter'
+                                               '/start/newsletter?flowName=newsletter&redirect_to=/setup/newsletterSetup?flow=newsletter'
                                        );

                                case 'newsletterSetup':
diff --git a/client/signup/config/flows-pure.js b/client/signup/config/flows-pure.js
index a88e688b82..53f5fd8cc6 100644
--- a/client/signup/config/flows-pure.js
+++ b/client/signup/config/flows-pure.js
@@ -113,7 +113,7 @@ export function generateFlows( {
                },
                {
                        name: 'newsletter',
-                       steps: [ 'domains', 'plans-newsletter' ],
+                       steps: [ 'user', 'domains', 'plans-newsletter' ],
                        destination: ( dependencies, localeSlug, goesThroughCheckout ) => {
                                return goesThroughCheckout
                                        ? `/setup/completingPurchase?flow=newsletter&siteSlug=${ dependencies.siteSlug }`
@@ -129,7 +129,7 @@ export function generateFlows( {
                },
                {
                        name: 'link-in-bio',
-                       steps: [ 'domains', 'plans-link-in-bio' ],
+                       steps: [ 'user', 'domains', 'plans-link-in-bio' ],
                        destination: ( dependencies ) =>
                                `/setup/completingPurchase?flow=link-in-bio&siteSlug=${ encodeURIComponent(
                                        dependencies.siteSlug

```

This might be better to leave to y'all since I imagine you have a better idea of how we're stretching our usage of these two frameworks.

**Comment by alshakero** [comment]
Interesting! Tony is dealing with this here https://github.com/Automattic/wp-calypso/pull/66950. We moved away from altering the flow name because I imagine it has the potential of creating unexpected behavior.

The user flow is technically part of onboarding, not LiB or Newsletter. So we created new param called variationName to avoid messing with that. Would love to hear your thoughts on this.

**Comment by ebinnion** [comment]
Using `variationName` makes some sense to me so that it's more clear that `flowName` doesn't override the `flowName` that gets chosen in the code.

But, we should make sure that we the user account gets created it's tracked back to the newsletter and link in bio flows.

**Comment by ebinnion** [comment]
To be clear, I don't think my `diff` is the answer. I was more so sharing it to demonstrate that the `signup_flow_name` seems to be derived from the flow's slug and that when I added the `user` step to the tailored flows, I got the expected flow slug. 

When we send users into the `account` flow, we just get the `account` slug as `signup_flow_name`.

I see a couple of ways forward currently:

- We duplicate the `account` flow and create something like `account-newsletter` and `account-link-in-bio` and then send users into that from Stepper
- We append the `variationName` to the `flowName` that gets passed to `SignupForm` [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/user/index.jsx#L457)

I think the first option is cleaner in terms of being expected for the framework. I think the latter option is quicker and makes some sense.

This does make me want to not have multiple frameworks for onboarding though :)
